### ISSUE TITLE
PyMJCF recursive include tags relative to base model

### ISSUE THREAD

**Issue by guyazran** [deepmind/dm_control#418]
PyMJCF does not support nested include tags the same way as the MuJoCo compiler. No matter how deep the include tag is, MuJoCo expects all included models and their assets to be given as paths to the corresponding file **relative to the input file parent directory**. However, the current implementation of PyMJCF checks paths relative to the current file's parent directory, which MuJoCo does not support.

In the attached [scenes.zip](https://github.com/deepmind/dm_control/files/12268186/scenes.zip) file are 3 scenes.

- my_scene1/scene.xml includes a file in an inner directory that includes another file in yet another inner directory. The internal include tag uses a path relative to the top-level model directory (my_scene1/). This is supported by MuJoCo but PyMJCF fails to parse the scene:
```python
>>> mjcf.Physics.from_xml_path('my_scene1/scene.xml')  # parse using mujoco directly
<dm_control.mjcf.physics.Physics object at 0x7fd0d191b0d0>

>>> mjcf.Physics.from_mjcf_model(mjcf.from_path('my_scene1/scene.xml'))  # parse with PyMJCF then convert to physics
FileNotFoundError: [Errno 2] No such file or directory: 'my_scene1/inner/inner/inner-inner/include.xml'
```
- my_scene2/scene.xml is the same as my_scene1/scene.xml but the inner include path is given relative to the current model directory (my_scene2/inner/). PyMJCF supports this but it does not compile natively using MuJoCo.
```python
>>> mjcf.Physics.from_xml_path('my_scene2/scene.xml')  # parse using mujoco directly
ValueError: XML Error: Could not open file 'my_scene2/inner-inner/include.xml'
Element 'include', line 0

>>> mjcf.Physics.from_mjcf_model(mjcf.from_path('my_scene2/scene.xml'))  # parse with PyMJCF then convert to physics
<dm_control.mjcf.physics.Physics object at 0x7fc61f1a6700>
```
- my_scene3/scene.xml specifies mesh and texture files in an included file to show that MuJoCo expects these files to be specified relative to the top-level model directory as well.
```python
>>> mjcf.Physics.from_xml_path('my_scene3/scene.xml')  # parse using mujoco directly
<dm_control.mjcf.physics.Physics object at 0x7fc61f1a6f10>

>>> mjcf.Physics.from_mjcf_model(mjcf.from_path('my_scene2/scene.xml'))  # parse with PyMJCF then convert to physics
FileNotFoundError: Line 4: error while parsing element <mesh>: during initialization of attribute 'file' of element <mesh>: [Errno 2] No such file or directory: 'my_scene3/inner/meshes/banana.stl'
```

In conclusion, models with nested include tags or external assets in included files are parsed differently with MuJoCo than with PyMJCF.
